mlflow/mlflow
# Build out predict step

Signed-off-by: Jerry Liang <jerry.liang@databricks.com>
## What changes are proposed in this pull request?

Added config parsing and basic Spark UDF scoring to the MLP batch scoring predict step.

Future work:
* Tests for scoring with registered models
* Tests on the scored dataframe itself. Make sure that reasonable predictions are actually made.
* Make use of environment restoration for Spark UDF. (I encountered some weird bugs when using it during testing, so I decided to defer it to future work)
* Build out the output writing functionality of the predict step and corresponding steps.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

I ran the tests locally and they passed.



## Does this PR change the documentation?

- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

## Release Notes

### Is this a user-facing change?


(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [x] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates

Interface

Language

Integrations


<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:
